lambda/translquote.ml

```diff
@@ -0,0 +1,3144 @@
+
+    let false_ = use "Identifier.Constructor" "false_"
+
+    let true_ = use "Identifier.Constructor" "true_"
+
+    let void = use "Identifier.Constructor" "void"
+
+    let nil = use "Identifier.Constructor" "nil"
+
+    let cons = use "Identifier.Constructor" "cons"
+
+    let none = use "Identifier.Constructor" "none"
+
+    let some = use "Identifier.Constructor" "some"
+
+    let match_failure = use "Identifier.Constructor" "match_failure"
+
+    let out_of_memory = use "Identifier.Constructor" "out_of_memory"
+
+    let invalid_argument = use "Identifier.Constructor" "invalid_argument"
+
+    let failure = use "Identifier.Constructor" "failure"
+
+    let not_found = use "Identifier.Constructor" "not_found"
+
+    let sys_error = use "Identifier.Constructor" "sys_error"
+
+    let end_of_file = use "Identifier.Constructor" "end_of_file"
+
+    let division_by_zero = use "Identifier.Constructor" "division_by_zero"
+
+    let stack_overflow = use "Identifier.Constructor" "stack_overflow"
+
+    let sys_blocked_io = use "Identifier.Constructor" "sys_blocked_io"
+
+    let assert_failure = use "Identifier.Constructor" "assert_failure"
+
+    let undefined_recursive_module =
+      use "Identifier.Constructor" "undefined_recursive_module"
+  end
+
+  module Field = struct
+    type s = lambda
+
+    type t' = s lazy_t
+
+    type t = s lam
+
+    let wrap = inject_force
+
+    let dot loc a1 a2 =
+      apply2 "Identifier.Field" "dot" loc (extract a1) (string loc a2)
+  end
+end
+
+module Label : sig
+  type s
+
+  type t' = s lazy_t
+
+  type t = s lam
+
+  val wrap : t' -> t
+
+  module Nonoptional : sig
+    type s
+
+    type t' = s lazy_t
+
+    type t = s lam
+
+    val wrap : t' -> t
+
+    val no_label : t'
+
+    val labelled : Location.t -> string -> t'
+  end
+
+  val no_label : t'
+
+  val labelled : Location.t -> string -> t'
+
+  val optional : Location.t -> string -> t'
+end = struct
+  type s = lambda
+
+  type t' = s lazy_t
+
+  type t = s lam
+
+  let wrap = inject_force
+
+  module Nonoptional = struct
+    type s = lambda
+
+    type t' = s lazy_t
+
+    type t = s lam
+
+    let wrap = inject_force
+
+    let no_label = use "Label.Nonoptional" "no_label"
+
+    let labelled loc a1 =
+      apply1 "Label.Nonoptional" "labelled" loc (string loc a1)
+  end
+
+  let no_label = use "Label" "no_label"
+
+  let labelled loc a1 = apply1 "Label" "labelled" loc (string loc a1)
+
+  let optional loc a1 = apply1 "Label" "optional" loc (string loc a1)
+end
+
+module Module_type : sig
+  type s
+
+  type t' = s lazy_t
+
+  type t = s lam
+
+  val wrap : t' -> t
+
+  val ident : Location.t -> Identifier.Module_type.t -> t'
+
+  val of_string : Location.t -> string -> t'
+end = struct
+  type s = lambda
+
+  type t' = s lazy_t
+
+  type t = s lam
+
+  let wrap = inject_force
+
+  let ident loc a1 = apply1 "Module_type" "of_string" loc (extract a1)
+
+  let of_string loc a1 = apply1 "Module_type" "ident" loc (string loc a1)
```

Comment:
These feel like they're the wrong way around.

Reply:
Whoops! Well spotted. Means we need more tests of module types.